### DISCUSSION TITLE
Same message sent more than one time

### DISCUSSION THREAD

**Discussion by bilo2000** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#69] [General]
Hi,

i am fine with the solution, everything works perfect, except for sometime when i publish the solution then for the first message it got sent twice or more, the same message.

What is the reason, is it because i am using the asynch methods with wait or what? 

how can i prevent it

Regards

**Comment by gabrieldwight** [comment]
Maybe network delay on your end. How have you implemented the async methods in your project

**Comment by bilo2000** [comment]
Hi,
here is the code

        private async Task SendWhatsAppListMessage(string textSenderMobileNumber, string textSenderSelection,string textSenderMessageID)
        {
            using var conn = XXXXXXXX;

            conn.Open();
            var res = conn.Query<WhatsappExchanger>("exec my query goes here");
            var header = conn.QueryFirstOrDefault<string>("exec my query goes here");
            var body = conn.QueryFirstOrDefault<string>("exec my query goes here");
            var button = conn.QueryFirstOrDefault<string>("exec my query goes here");
            var footer = conn.QueryFirstOrDefault<string>("exec my query goes here");
            conn.Close();

           InteractiveListMessageRequest interactiveListMessage = new InteractiveListMessageRequest();
            interactiveListMessage.To = textSenderMobileNumber;
            interactiveListMessage.Interactive = new InteractiveListMessage();
            interactiveListMessage.Interactive.Header = new Header();
            interactiveListMessage.Interactive.Header.Type = "text";
            interactiveListMessage.Interactive.Header.Text = header ?? " No Header";

            interactiveListMessage.Interactive.Body = new ListBody();
            interactiveListMessage.Interactive.Body.Text = body ?? " No Brief ";

            interactiveListMessage.Interactive.Footer = new Footer();
            interactiveListMessage.Interactive.Footer.Text = footer ?? "No Step Description";

            interactiveListMessage.Interactive.Action = new ListAction();
            interactiveListMessage.Interactive.Action.Button = button ?? " Send ";
            interactiveListMessage.Interactive.Action.Sections = new List<Section>()
                    {
                        new Section()
                        {
                            Title = header ?? " No Step Label ",
                            Rows = res.Select(x => new Row()
                                {
                                    Id = x.ID.ToString(),
                                    Title = x.Label.ToString(),
                                    Description = ""
                                }).ToList(),
                        },
                    };

            await _whatsAppBusinessClient.SendInteractiveListMessageAsync(interactiveListMessage);
        }

**Reply by gabrieldwight** [reply]
The above code looks okay no issues. If you are doing a desktop project you will have to profile your app to check if your method is executed twice from one of the control event handlers where you have implemented in the code. On web it is not sending the message more than twice.

**Reply by bilo2000** [reply]
Actually i getting this only from web after deployment, for the first message only, after that, the issue never appear, it seems when IIS boots the solution everytime the Web.config file got editted

**Reply by gabrieldwight** [reply]
You will need to check what is happening on the web.config file on the syntax generated by IIS.

**Reply by theCJMan** [reply]
I have now seen a similar issue, but still running (debug mode) on my desktop.
It seems the sync between WhatsApp and my PC got confused or hanged, or something. I then stopped the code, restarted the code and somewhere the code is getting an error, BUT it is not going into the generic catch all.. it just bounces out. 

When I step slowly through the code, it seems it is executing lines of code all over the show, possibly an async that did not complete correctly and then on next run it continues... not sure..
